### ISSUE TITLE
[feature-request] Integrate "Tor" or similar to improve "PrivateSend" anonymity

### ISSUE THREAD

**Issue by Technologov** [dashpay/dash#1436]
Dash PrivateSend does a coin mixing... but afterwards you submit a transaction and broadcast it from your own IP address (assuming Dash Core wallet), right ?
So people can track you by IP address. (both original and mixed coins)
To send the mixed coins, you need to broadcast a transaction from your wallet. At this step, IP address becomes public. (except if you use a VPN or TOR)

This feature request is about integrating "Tor" client directly into "Dash Core" wallet, or alternatively, to develop similar functionality based on our Dash Masternode network.

Not urgent, but nice-to-have. This will improve anonymity of Dash, at a low cost.

-Technologov, 16.04.2017

**Comment by Technologov** [comment]
Jenya said "Integrating Tor in Dash would be a very bad idea. It would make any legal acceptance of Dash in e.g. Russia virtually impossible, no legal business in Russia would touch anything Tor related:

https://www.theregister.co.uk/2017/04/13/tor_loses_a_node_in_russia_after_activists_arrest_in_moscow/

https://blog.torservers.net/20170414/tor-exit-node-operator-arrested-in-russia-a-solidarity-tor-relay-challenge-launched.html". 

Basically if Tor is to be integrated, it should be strictly optional, as for users  in oppressive regimes it can hurt.

**Comment by tbbw** [comment]
I agree with Jenya... bad to ship in by default due to the political issues Tor has.
It is fairly easy to get dash to run thru tor as it is so i don't see why it would need to be included as a default feature.

**Comment by jihanwu** [comment]
Dash is already very anonymous, PBOC will not allow Tor coins through GFW. I think Tor is bad for usability.

**Comment by UdjinM6** [comment]
There are no plans to integrate Tor modules into the software itself but you can use `-onion` option to use separate SOCKS5 proxy to reach peers via Tor hidden services. Closing
